### PR TITLE
Option to specify RemoteTask arguments by keyword, position, dictionary, or position

### PR DESCRIPTION
**Pull request by David-Simonetti-ND**
When creating a RemoteTask, one can specify arguments by keyword as before, but now there is support for positional arguments and passing in a dictionary containing keyword arguments (pass in the dictionary as work_queue_argument_dict). The user can also mix and match these arguments, passing some arguments by keyword and some by position.

### CODE REVIEW

File: work_queue/src/bindings/python3/work_queue.binding.py [hunk 1]
```diff
@@ -2073,10 +2073,16 @@ class RemoteTask(Task):
     # @param coprocess  The name of the coprocess which has the function you wish to execute. The coprocess should have a name() method that returns this
     # @param
     # @param command    The shell command line to be exected by the task.
-    # @param kwargs	    keyword arguments used in function to be executed by task. An optional kwarg exec_method can be specified to one of fork, thread, and direct to choose between those three methods of execution
-    def __init__(self, fn, coprocess, **kwargs):
+    # @param args       positional arguments used in function to be executed by task. Can be mixed with kwargs
+    # @param kwargs	    keyword arguments used in function to be executed by task. An optional kwarg exec_method can be specified to one of fork, thread, and direct to choose between those three methods of execution. Another optional kwarg is work_queue_argument_dict, which contains a dictionary of keyword arguments for the remote function
+    def __init__(self, fn, coprocess, *args, **kwargs):
```
Comment (btovar) [inline work_queue/src/bindings/python3/work_queue.binding.py:2078]: What do you think of adding a `specify_exec_method()` to `RemoteTask`?   I don't like that the kwarg `exec_method` gets mixed with the args to the function.
Such method may also simplify the code needed to remove the `exec_method` key from the kwarg dict.

**Review by btovar — CHANGES_REQUESTED** [review]

**Comment by David-Simonetti-ND** [comment]
Changed RemoteTask so that the event is only transferred to infile once the task is submitted to the queue, because otherwise the event would be turned into an infile once the task was created and would not allow any changes to be made (since the file was already created). This allows two new functions, specify_argument_dictionary and specify_exec_method which allow the user to set a dictionary as argument and change the exec_method on the network side. The exec_method is now no longer passed with the kwargs of the function.

```
t = wq.RemoteTask("my_sum", "my_coprocess_example")
t.specify_argument_dictionary(argument_dict={"a":5, "b":5})
t.specify_exec_method("fork")
queue.submit(t)
t = wq.RemoteTask("my_sum", "my_coprocess_example", a=2, b=3)
queue.submit(t)
```

File: work_queue/src/bindings/python3/work_queue.binding.py [hunk 2]
```diff
@@ -1734,6 +1734,8 @@ def tune(self, name, value):
     # @param self   Reference to the current work queue object.
     # @param task   A task description created from @ref work_queue::Task.
     def submit(self, task):
+        if "_event" in task.__dict__:
```
Comment (btovar) [inline work_queue/src/bindings/python3/work_queue.binding.py:1737]: Instead use something like: `if isinstance(task, RemoteTask):`

File: work_queue/src/bindings/python3/work_queue.binding.py [hunk 3]
```diff
@@ -2073,12 +2075,33 @@ class RemoteTask(Task):
...
-    # @param kwargs	    keyword arguments used in function to be executed by task. An optional kwarg exec_method can be specified to one of fork, thread, and direct to choose between those three methods of execution
-    def __init__(self, fn, coprocess, **kwargs):
+    # @param args       positional arguments used in function to be executed by task. Can be mixed with kwargs
+    # @param kwargs	    keyword arguments used in function to be executed by task. 
+    def __init__(self, fn, coprocess, *args, **kwargs):
         Task.__init__(self, fn)
-        event = json.dumps(kwargs)
-        Task.specify_buffer(self, event, "infile")
+        self._event = {}
+        self._event["work_queue_kwargs"] = kwargs
```
Comment (btovar) [inline work_queue/src/bindings/python3/work_queue.binding.py:2083]: Rename these ones to  `fn_args` and `fn_kwargs`. They relate to the function, not wq.

File: work_queue/src/bindings/python3/work_queue.binding.py [hunk 4]
```diff
@@ -2073,12 +2075,33 @@ class RemoteTask(Task):
...
+        self._event["work_queue_positional_args"] = args
         Task.specify_coprocess(self, coprocess)
+    ##
+    # Specify function arguments as a dictionary argument. This function overrides and kwargs and args passed to the task previously
+    # @param self             Reference to the current remote task object
+    # @param *args            Positonal args to passed to the function 
+    # @param argument_dict    A dictionary that contains arguments by keyword. The key is the name of the argument and the value is the value that should be passed to the argument
+    def specify_argument_dictionary(self, *args, argument_dict=None):
```
Comment (btovar) [inline work_queue/src/bindings/python3/work_queue.binding.py:2091]: `specify_fn_args(args=None, kwargs=None)` as it does more than the dict. Use like: `specify_fn_args([1,2,3], {"a": 2})`

**Review by btovar — CHANGES_REQUESTED** [review]

**Comment by btovar** [comment]
Ready to merge?

**Comment by David-Simonetti-ND** [comment]
Yup! Ready to merge

**Pull request merged**